### PR TITLE
Tools : Added custom ROS-2 service definition for switching drive modes

### PR DESCRIPTION
**Pull request by arshPratap**
Added custom ROS-2 service definition for switching drive modes.
Service definition used from [here](https://github.com/arshPratap/ardupilot_ros2/blob/main/ap_custom_services/srv/ModeSelect.srv)
@Ryanf55 @srmainwaring @tridge

### CODE REVIEW

File: Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Rover  : https://github.com/ArduPilot/ardupilot/blob/master/Rover/mode.h#L12
+# Plane  : https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/mode.h#L22
+
+int8 mode
+---
+# status    : True if the request for mode switch was successful, False otherwise
+# curr_mode : Returns the code for the current drive/flight mode , after the processing the request
+
+bool status
+int8 curr_mode
```
Comment (Ryanf55) [inline Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv:15]: Please add trailing newline so you `cat` the file without messing up your terminal.

File: Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+# This service requests the vehicle to switch its drive/flight mode
+
+# mode : Set the value to the drive/flight mode to be used
+# Copter : https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/mode.h#L14 
+# Rover  : https://github.com/ArduPilot/ardupilot/blob/master/Rover/mode.h#L12
+# Plane  : https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/mode.h#L22
```
Comment (Ryanf55) [inline Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv:7]: Is there a more maintainable way to link the modes that won't get out of date when the line numbers change?

**Review by Ryanf55 — CHANGES_REQUESTED** [review]

File: Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv (hunk 2 above)
Comment (srmainwaring) [reply Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv:7]: Perhaps we could just drop the line numbers and provide a link to respective wikis for more detail?

Comment (srmainwaring) [inline Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv:7]:
```suggestion
# Copter : https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/mode.h 
# Rover  : https://github.com/ArduPilot/ardupilot/blob/master/Rover/mode.h
# Plane  : https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/mode.h
```

**Review by srmainwaring — CHANGES_REQUESTED** [review]

File: Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+int8 mode
```
Comment (tridge) [inline Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv:9]: int8 is signed? why not uint8 ?

File: Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv (hunk 1 above)
Comment (tridge) [inline Tools/ros2/ardupilot_msgs/srv/ModeSwitch.srv:15]: why signed?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by arshPratap** [comment]
@Ryanf55 @tridge @srmainwaring thanks for the review..have made the changes..Thanks..would like to know your thoughts

**Review by srmainwaring — APPROVED** [review]

**Comment by tridge** [comment]
@Ryanf55 looks good to me, you OK with the changes?

**Comment by tridge** [comment]
could use these links:
https://mavlink.io/en/messages/ardupilotmega.html#COPTER_MODE
https://mavlink.io/en/messages/ardupilotmega.html#PLANE_MODE
https://mavlink.io/en/messages/ardupilotmega.html#ROVER_MODE

**Comment by arshPratap** [comment]
> could use these links: https://mavlink.io/en/messages/ardupilotmega.html#COPTER_MODE https://mavlink.io/en/messages/ardupilotmega.html#PLANE_MODE https://mavlink.io/en/messages/ardupilotmega.html#ROVER_MODE

Thanks @tridge .. Have made the changes

**Review by Ryanf55 — APPROVED** [review]

**Pull request merged**